src/Core/Services/Cache/DabCacheService.cs

```diff
@@ -78,6 +78,35 @@ public DabCacheService(IFusionCache cache, ILogger<DabCacheService>? logger, IHt
         return result;
     }
 
+    /// <summary>
+    /// Attempts to fetch response from cache. If there is a cache miss, invoke executeQueryAsync Func to get a response
+    /// </summary>
+    /// <typeparam name="TResult">Response payload Type</typeparam>
+    /// <param name="executeQueryAsync">Func with a result of type TResult. Only executed after a cache miss.</param>
+    /// <param name="queryMetadata">Metadata used to create a cache key or fetch a response from the database.</param>
+    /// <param name="cacheEntryTtl">Number of seconds the cache entry should be valid before eviction.</param>
+    /// <returns>JSON Response</returns>
+    /// <exception cref="Exception">Throws when the cache-miss factory method execution fails.</exception>
+    public async ValueTask<TResult?> GetOrSetAsync<TResult>(Func<Task<TResult>> executeQueryAsync, DatabaseQueryMetadata queryMetadata, int cacheEntryTtl)
+    {
+
+        string cacheKey = CreateCacheKey(queryMetadata);
```

Comment:
Please either create a tracking issue for it, or make sure to send a followup PR.

Reply:
I can create a tracking item. Since I couldn't locate caching test cases for the SQL implementation, it would be preferable for the SQL team to handle any follow-up PR if they intend to implement this generic behavior. Integration tests can also be incorporated into the same PR for comprehensive coverage.